### ISSUE TITLE
The Implement Of HTTPMethodView Is Not A Good  Practice.

### ISSUE THREAD

**Issue by TaylorHere** [channelcat/sanic#267]
sorry,I'm lack of English.
```
def __call__(self, request, *args, **kwargs):
        handler = getattr(self, request.method.lower(), None)
        if handler:
            return handler(request, *args, **kwargs)
        raise InvalidUsage(
            'Method {} not allowed for URL {}'.format(
                request.method, request.url), status_code=405)
```
In those code ,when I handle the request, it's not the HTTPMethodView itself to  be instantiation,but the handler.
the handler just carry the request,but not the instance,so that I can't use the instance to store some message.
So, I think it's not the right practice of a "View based Class".
In View based Class we should treat the request as the instance itself,so that I can write the code base on view itself,and do not think about the request.

**Comment by seemethere** [comment]
@AntonDnepr?

**Comment by AntonDnepr** [comment]
@seemethere @TaylorHere class-based views was written in minimalistic style just to provide possibility to write class-based style code. I agree, instance should be added, because we need to have a possibility to call additional class methods in views. I'm not agree that we shouldn't have separate request as a parameter, but this is question to discuss if need.
We have an open PR which tries to implement this kind of approach
https://github.com/channelcat/sanic/pull/201/files
but there are several problems with it and on my opinion it's an overkill for now.

I think this should be mark as thing to do and I'll try to implement this (but I can't start earlier then next week, so if someone could do this faster - it would be cool)

**Comment by yoshiya0503** [comment]
If you don't mind, please reference Specific example in the following mini app.

`
from asyncio import sleep
from sanic import Sanic
from sanic.response import json
from sanic.views import HTTPMethodView

app = Sanic()

class Test(HTTPMethodView):

    def __init__(self):
        self.g = 0

    async def get(self, req):
        self.g += 10
        print(self.g)
        await sleep(0.1)
        print(self.g)
        # !!!  we expect self.g == 10 each context
        return json({'test': 'class test'})

app.add_route(Test(), '/class')

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=8000)
`

If we send some requests **concurrency** (using apache bench).

`
ab -c 10 -n 100 http://127.0.0.1:8000/class
`

g.self is equal to 1000, but actually, we expect g.self is equal to 10.

Oh gotcha!

We expect that **instance variables are always context local(like thread safe)**.
In the case of single thread non-blocking model (same as node.js), because of its feature,
**instantiation of class base handlers just once at the server startup** is not good practice.
(Same problem will also occur even if we use class method.)

[solutions]
instantiation of class base handlers should be just create every requests, like tornado web framework.
(it seem so difficult to implement using existing function base handlers...)

**Comment by AntonDnepr** [comment]
@yoshiya0503 great example, thank you! Will take this in consideration while fixing.

**Comment by seemethere** [comment]
Closed by #274
